src/server/qgsserviceregistry.h

```diff
@@ -71,16 +73,53 @@ class SERVER_EXPORT QgsServiceRegistry
      * This method is intended to  be called by modules for registering
      * services. A module may register multiple services.
      *
-     * The registry gain ownership of services and will call 'delete' on cleanup
+     * The registry takes ownership of services and will call 'delete' on cleanup
      *
      * \param service a QgsService to be registered
      */
     void registerService( QgsService *service SIP_TRANSFER );
 
+    /**
+     * Registers the QgsServerApi \a api
+     *
+     * The registry takes ownership of services and will call 'delete' on cleanup
+     * \since QGIS 3.10
+     */
+    void registerApi( QgsServerApi *api SIP_TRANSFER );
+
+    /**
+     * Unregister API from its name and version
+     *
+     * \param name the name of the service
+     * \param version (optional) the specific version to unload
+     * \returns the number of APIs unregistered
+     *
+     * If the version is not specified then all versions from the specified API
+     * are unloaded
+     * \since QGIS 3.10
+     */
+    int unregisterApi( const QString &name, const QString &version = QString() );
+
+    /**
+     * Search the API register for an API matching the \a request and returns a (possibly NULL) pointer to it.
+     * \since QGIS 3.10
+     */
+    QgsServerApi *getApiForRequest( const QgsServerRequest &request ) const SIP_SKIP;
```

Comment:
I'm not a big fan of getters named `getBlaBla()`. Do you think we could renamed it like `apiForRequest()` instead? I know we have a contrasting situation in QGIS (with or without `get()`), but it seems to me that it would be more homogeneous with the server code.

But I don't have a strong opinion on the matter :).